pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller_test.go

```diff
@@ -522,6 +529,88 @@ var _ = Describe("Reconcile Accessibility Logic", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(result).To(Equal(reconcile.Result{RequeueAfter: 0}))
 	})
+
+	It("should add vm name and storage policy reservation labels to PVC when present on CR", func() {
+		// Test the getPersistentVolumeClaimSpec function directly
+		instance := &cnsregistervolumev1alpha1.CnsRegisterVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-volume-with-labels",
+				Namespace: "test-ns",
+				Labels: map[string]string{
+					cnsoperatortypes.LabelVirtualMachineName:           "test-vm-name",
+					cnsoperatortypes.LabelStoragePolicyReservationName: "test-storage-policy-reservation-name",
+				},
+			},
+			Spec: cnsregistervolumev1alpha1.CnsRegisterVolumeSpec{
+				PvcName:    "test-pvc-with-labels",
+				VolumeID:   "dummy-volume-id",
+				AccessMode: "ReadWriteOnce",
+			},
+		}
+
+		// Test getPersistentVolumeClaimSpec function directly
+		pvcSpec, err := getPersistentVolumeClaimSpec(ctx, "test-pvc", "test-ns", 1024,
+			"test-storage-class", corev1.ReadWriteOnce, "test-pv", nil, instance)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvcSpec).NotTo(BeNil())
+		Expect(pvcSpec.Labels).To(HaveKeyWithValue(cnsoperatortypes.LabelVirtualMachineName, "test-vm-name"))
+		Expect(pvcSpec.Labels).To(HaveKeyWithValue(cnsoperatortypes.LabelStoragePolicyReservationName,
+			"test-storage-policy-reservation-name"))
+	})
+
+	It("should not add labels to PVC when not present on CR", func() {
+		// Test the getPersistentVolumeClaimSpec function directly without labels
+		instance := &cnsregistervolumev1alpha1.CnsRegisterVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-volume-without-labels",
+				Namespace: "test-ns",
+				// No labels set
+			},
+			Spec: cnsregistervolumev1alpha1.CnsRegisterVolumeSpec{
+				PvcName:    "test-pvc-without-labels",
+				VolumeID:   "dummy-volume-id",
+				AccessMode: "ReadWriteOnce",
+			},
+		}
+
+		// Test getPersistentVolumeClaimSpec function directly
+		pvcSpec, err := getPersistentVolumeClaimSpec(ctx, "test-pvc", "test-ns", 1024,
+			"test-storage-class", corev1.ReadWriteOnce, "test-pv", nil, instance)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvcSpec).NotTo(BeNil())
+		Expect(pvcSpec.Labels).NotTo(HaveKey(cnsoperatortypes.LabelVirtualMachineName))
+		Expect(pvcSpec.Labels).NotTo(HaveKey(cnsoperatortypes.LabelStoragePolicyReservationName))
+	})
+
+	It("should add only vm name label when only that label is present on CR", func() {
```

Comment:
Here we need to update description to "should NOT add only vm name label when only that label is present on CR"